sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs

```diff
@@ -93,5 +95,25 @@ public static IWebJobsBuilder AddServiceBus(this IWebJobsBuilder builder, Action
             builder.Services.AddSingleton<ServiceBusClientFactory>();
             return builder;
         }
+
+        public static IWebJobsBuilder AddServiceBusScaleForTrigger(this IWebJobsBuilder builder, TriggerMetadata triggerMetadata)
```

Comment:
Where is TriggerMetadata defined?
Does this method need to be public?

Reply:
Also, is the fact that this is for the trigger important to include in the method name? What about AddServiceBusScaler or AddServiceBusScaling?

Reply:
We are going to call the method per trigger in ScaleController. See an example in the E2E test:
https://github.com/Azure/azure-sdk-for-net/pull/34557/files#diff-2f37d5be90529dcbb846e2db09fd6d906bec85a72a6f637e8fd8d429d017c4aeR115

This is the only public interface change in the PR and we need it to call from ScaleController. The method adds to DI scale entities for specific trigger.  We do not expect function app customers to call this method.

Reply:
I think this is the type - https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Scale/TriggerMetadata.cs#L13

It is unusual for us to add public surface that references a non-framework type from another library that is not part of the Azure SDK. Is the TriggerMetadata type already GA?

Reply:
Another comment - why not just take the AzureComponentFactory as a parameter rather than needing to fish it out of the properties dictionary in a non-typed way?

Reply:
`TriggerMetadata` is already GA (we had released the web jobs sdk 3.0.37).
We did not want to introduce new public types for each trigger, all the extensions will have the same mew public method:
`Add{**EXTENSION_NAME**}ScaleForTrigger(this IWebJobsBuilder builder, TriggerMetadata triggerMetadata)`

We considered passing AzureComponentFactory  as parameter and decided to use the properties collection as it give us more flexibility if you want to pass some other data in future.